pkg/sql/sem/builtins/builtins.go

```diff
@@ -5862,6 +5862,95 @@ table's zone configuration this will return NULL.`,
 			Volatility: tree.VolatilityVolatile,
 		},
 	),
+
+	"crdb_internal.serialize_session": makeBuiltin(
+		tree.FunctionProperties{
+			Category: categorySystemInfo,
+		},
+		tree.Overload{
+			Types:      tree.ArgTypes{},
+			ReturnType: tree.FixedReturnType(types.Bytes),
+			Fn: func(evalCtx *tree.EvalContext, args tree.Datums) (tree.Datum, error) {
+				if !evalCtx.TxnImplicit {
+					return nil, pgerror.Newf(
+						pgcode.InvalidTransactionState,
+						"cannot serialize a session which is inside a transaction",
+					)
+				}
+
+				sd := evalCtx.SessionData
+				if sd == nil {
+					return nil, pgerror.Newf(
+						pgcode.InvalidTransactionState,
+						"no session is active",
+					)
+				}
+
+				if len(sd.DatabaseIDToTempSchemaID) > 0 {
+					return nil, pgerror.Newf(
+						pgcode.InvalidTransactionState,
+						"cannot serialize session with temporary schemas",
+					)
+				}
+
+				var m sessiondatapb.MigratableSession
+				m.SessionData = sd.SessionData
+				sessiondata.MarshalNonLocal(sd, &m.SessionData)
+				m.LocalOnlySessionData = sd.LocalOnlySessionData
+
+				b, err := m.Marshal()
+				if err != nil {
+					return nil, err
+				}
+				return tree.NewDBytes(tree.DBytes(b)), nil
+			},
+			Info:       `This function serializes the variables in the current session.`,
+			Volatility: tree.VolatilityVolatile,
+		},
+	),
+
+	"crdb_internal.deserialize_session": makeBuiltin(
+		tree.FunctionProperties{
+			Category: categorySystemInfo,
+		},
+		tree.Overload{
+			Types:      tree.ArgTypes{{"session", types.Bytes}},
+			ReturnType: tree.FixedReturnType(types.Bool),
+			Fn: func(evalCtx *tree.EvalContext, args tree.Datums) (tree.Datum, error) {
+				if !evalCtx.TxnImplicit {
+					return nil, pgerror.Newf(
+						pgcode.InvalidTransactionState,
+						"cannot deserialize a session which is inside a transaction",
+					)
+				}
+
+				var m sessiondatapb.MigratableSession
+				if err := m.Unmarshal([]byte(tree.MustBeDBytes(args[0]))); err != nil {
+					return nil, pgerror.WithCandidateCode(
+						errors.Wrapf(err, "error deserializing session"),
+						pgcode.InvalidParameterValue,
+					)
+				}
+				sd, err := sessiondata.UnmarshalNonLocal(m.SessionData)
+				if err != nil {
+					return nil, err
+				}
+				sd.SessionData = m.SessionData
+				sd.LocalUnmigratableSessionData = evalCtx.SessionData.LocalUnmigratableSessionData
+				sd.LocalOnlySessionData = m.LocalOnlySessionData
+				if sd.User().Normalized() != evalCtx.SessionData.User().Normalized() {
+					return nil, pgerror.Newf(
+						pgcode.InsufficientPrivilege,
+						"can only serialize matching users",
+					)
+				}
+				*evalCtx.SessionData = *sd
+				return tree.MakeDBool(true), nil
+			},
+			Info:       `This function serializes the variables in the current session.`,
```

Comment:
I think this is Info is for `serialize` not `deserialize`

Reply:
you sunk my battleship
(good catch)